server/src/main/java/org/jboss/as/server/services/net/LocalDestinationOutboundSocketBindingAddHandler.java

```diff
@@ -71,6 +79,30 @@ protected void performRuntime(final OperationContext context, final ModelNode op
 
         final PathAddress address = PathAddress.pathAddress(operation.get(OP_ADDR));
         final String outboundSocketName = address.getLastElement().getValue();
+
+        context.addStep(new OperationStepHandler() {
+            @Override
+            public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+                final String currentSocketBindingGroupName = address.getParent().getLastElement().getValue();
+                final Set<ResourceEntry> children = context.readResourceFromRoot(PathAddress.EMPTY_ADDRESS)
+                        .getChildren(ModelDescriptionConstants.SOCKET_BINDING_GROUP);
+                for (Resource.ResourceEntry re : children) {
+                    if (re.getName().equalsIgnoreCase(currentSocketBindingGroupName)) {
```

Comment:
2) of 2)

Assuming the analysis of 1) shows this can be limited to non-domain servers, then L87-90 seems overly complex.

Resource re = context.readResourceFromRoot(PathAddress.pathAddress(SOCKET_BINDING_GROUP, currentSocketBindingGroupName)) should give you the only resource that's relevant.